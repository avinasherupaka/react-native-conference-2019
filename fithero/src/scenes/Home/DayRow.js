/* @flow */

import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import { Text } from 'react-native-paper';

import DayItem from './DayItem';
import {
  dateToString,
  dateToWorkoutId,
  getDatePrettyFormat,
  getToday,
  isSameDay,
} from '../../utils/date';
import type { WorkoutSchemaType } from '../../database/types';

type Props = {
  currentWeek: Array<Date>,
  onDaySelected: string => void,
  selected: string,
  workouts: { [date: string]: WorkoutSchemaType },
};

class DayRow extends React.Component<Props> {
  _onDaySelected = dateString => {
    this.props.onDaySelected(dateString);
  };

  _renderDays() {
    return this.props.currentWeek.map(d => {
      const isSelected = isSameDay(d, this.props.selected);
      const dateString = dateToWorkoutId(d);
      return (
        <DayItem
          key={dateString}
          dateString={dateString}
          isSelected={isSelected}
          onDaySelected={this._onDaySelected}
          isWorkout={!!this.props.workouts[dateString]}
        />
      );
    });
  }

  render() {
    return (
      <React.Fragment>
        <Text style={styles.title}>
          {getDatePrettyFormat(this.props.selected, dateToString(getToday()))}
        </Text>
        <View style={styles.row}>{this._renderDays()}</View>
      </React.Fragment>
    );
  }
}

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 8,
    paddingBottom: 4,
  },
  title: {
    fontSize: 15,
    paddingVertical: 8,
    paddingHorizontal: 16,
  },
});

export default DayRow;
