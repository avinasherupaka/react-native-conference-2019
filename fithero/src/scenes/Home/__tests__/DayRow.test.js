/* @flow */

import * as React from 'react';
import { render } from 'react-native-testing-library';
import moment from 'moment';

import DayRow from '../DayRow';
import {
  getCurrentWeek,
  setMomentFirstDayOfTheWeek,
} from '../../../utils/date';

describe('DayRow', () => {
  const getRender = (day: string) => {
    const currentWeek = getCurrentWeek(moment(day).startOf('day'));
    return render(
      <DayRow
        currentWeek={currentWeek}
        workouts={{}}
        onDaySelected={jest.fn()}
        selected={day}
      />
    );
  };

  setMomentFirstDayOfTheWeek('en', 0, true);
  const { toJSON: mondayJSON } = getRender('2019-03-18T00:00:00.000Z');

  it('day row starts on Monday or Sunday', () => {
    setMomentFirstDayOfTheWeek('en', 1, true);
    const { toJSON: sundayJSON } = getRender('2019-03-17T00:00:00.000Z');

    // $FlowFixMe
    expect(mondayJSON()).toMatchDiffSnapshot(sundayJSON(), {
      contextLines: 0,
      stablePatchmarks: true,
    });
  });

  it('day row starts on Monday or Saturday', () => {
    setMomentFirstDayOfTheWeek('en', 6, true);
    const { toJSON: saturdayJSON } = getRender('2019-03-16T00:00:00.000Z');

    // $FlowFixMe
    expect(mondayJSON()).toMatchDiffSnapshot(saturdayJSON(), {
      contextLines: 0,
      stablePatchmarks: true,
    });
  });
});
