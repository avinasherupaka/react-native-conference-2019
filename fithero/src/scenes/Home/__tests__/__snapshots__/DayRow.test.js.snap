// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayRow day row starts on Monday or Saturday 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ --- --- @@
-     Monday, March 18
+     Saturday, March 16
@@ --- --- @@
-               false,
+               Object {
+                 "opacity": 1,
+               },
@@ --- --- @@
-             17
+             16
@@ --- --- @@
-             Sun
+             Sat
@@ --- --- @@
-               Object {
-                 "opacity": 1,
-               },
+               false,
@@ --- --- @@
-             18
+             17
@@ --- --- @@
-             Mon
+             Sun
@@ --- --- @@
-             19
+             18
@@ --- --- @@
-             Tue
+             Mon
@@ --- --- @@
-             20
+             19
@@ --- --- @@
-             Wed
+             Tue
@@ --- --- @@
-             21
+             20
@@ --- --- @@
-             Thu
+             Wed
@@ --- --- @@
-             22
+             21
@@ --- --- @@
-             Fri
+             Thu
@@ --- --- @@
-             23
+             22
@@ --- --- @@
-             Sat
+             Fri
`;

exports[`DayRow day row starts on Monday or Sunday 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ --- --- @@
-     Monday, March 18
+     Sunday, March 17
@@ --- --- @@
-             17
+             11
@@ --- --- @@
-             Sun
+             Mon
@@ --- --- @@
+               false,
@@ --- --- @@
-               Object {
-                 "opacity": 1,
-               },
@@ --- --- @@
-             18
+             12
@@ --- --- @@
-             Mon
+             Tue
@@ --- --- @@
-             19
+             13
@@ --- --- @@
-             Tue
+             Wed
@@ --- --- @@
-             20
+             14
@@ --- --- @@
-             Wed
+             Thu
@@ --- --- @@
-             21
+             15
@@ --- --- @@
-             Thu
+             Fri
@@ --- --- @@
-             22
+             16
@@ --- --- @@
-             Fri
+             Sat
@@ --- --- @@
-               false,
+               Object {
+                 "opacity": 1,
+               },
@@ --- --- @@
-             23
+             17
@@ --- --- @@
-             Sat
+             Sun
`;
